src/main/java/org/terasology/projsndwv/systems/ApiarySystem.java

```diff
@@ -15,23 +15,273 @@
+                        if (!success) {
+                            break;
+                        }
+                    }
+                }
+
+                queenBee.destroy();
+            } else {
+                queenBee.saveComponent(matedComponent);
+                delayManager.addDelayedAction(entity, LIFE_TICK_EVENT, TempBeeRegistry.getTickTimeFromGenome(queenBee.getComponent(GeneticsComponent.class).activeGenes.get(LOCUS_SPEED)));
+            }
+        }
+        else if (event.getActionId().equals(MATING_EVENT)) {
+            entity.removeComponent(ApiaryMatingComponent.class);
+
+            EntityManager entityManager = CoreRegistry.get(EntityManager.class);
+            if (entityManager == null) {
+                logger.error("No EntityManager in registry");
+                return;
+            }
+
+            EntityRef femaleBee = entity.getComponent(InventoryComponent.class).itemSlots.get(ApiarySystem.SLOT_FEMALE);
+            GeneticsComponent femaleGenetics = femaleBee.getComponent(GeneticsComponent.class);
+            EntityRef maleBee = entity.getComponent(InventoryComponent.class).itemSlots.get(ApiarySystem.SLOT_MALE);
+            femaleBee.addComponent(new MatedComponent(maleBee.getComponent(GeneticsComponent.class), TempBeeRegistry.getLifespanFromGenome(femaleGenetics.activeGenes.get(ApiarySystem.LOCUS_LIFESPAN)), entityManager));
+            BeeComponent beeComponent = femaleBee.getComponent(BeeComponent.class);
+            beeComponent.type = BeeComponent.BeeType.QUEEN;
+            femaleBee.saveComponent(beeComponent);
+            ItemComponent itemComponent = femaleBee.getComponent(ItemComponent.class);
+            itemComponent.icon = TempBeeRegistry.getTextureRegionAssetForSpeciesAndType(femaleBee.getComponent(GeneticsComponent.class).activeGenes.get(ApiarySystem.LOCUS_SPECIES), 2);
+            femaleBee.saveComponent(itemComponent);
+            femaleBee.saveComponent(TempBeeRegistry.getDisplayNameComponentForSpeciesAndType(femaleBee.getComponent(GeneticsComponent.class).activeGenes.get(ApiarySystem.LOCUS_SPECIES), 2));
+            maleBee.destroy();
+
+            DelayManager delayManager = CoreRegistry.get(DelayManager.class);
+            if (delayManager == null) {
+                logger.error("No DelayManager in registry");
+                return;
+            }
+
+            delayManager.addDelayedAction(entity, ApiarySystem.LIFE_TICK_EVENT, TempBeeRegistry.getTickTimeFromGenome(femaleGenetics.activeGenes.get(ApiarySystem.LOCUS_SPEED)));
+        }
+    }
+
+    private Genome getGenome() {
+        if (genome == null) {
+            genome = new Genome(4, worldGenerator.getWorldSeed().hashCode());
+
+            GeneticsComponent beeCComponent = new GeneticsComponent(4);
+
+            beeCComponent.activeGenes.add(2); // TODO: (Soundwave) This is bad, both because the size isn't ensured, and because it's messy.
```

Comment:
Is this to be addressed in a follow PR or this one?

Reply:
Actually it's unfortunately potentially out of scope at the moment. Further discussion will be appropriate this weekend.